client/src/services/notificationCenter.ts

```diff
@@ -10,92 +14,123 @@ export enum NotificationLevel {
   Error = 'ERROR',
 }
 
-export interface Notification {
-  id: string,
-  title?: string,
-  message: string,
-  level: NotificationLevel,
-  read: boolean,
-  time: DateTime,
-  data: object,
+export class Notification {
+  id: string;
+  time: DateTime;
+  read: boolean;
+  message: string;
+  level: NotificationLevel;
+  data: object;
+  title?: string;
+  constructor({
+    message,
+    level,
+    data,
+    title,
+  }: {
+    message: string,
+    level: NotificationLevel,
+    data: object,
+    title?: string,
+  }) {
+    this.id = uuid4();
+    this.time = DateTime.now();
+    this.read = false;
+    this.message = message;
+    this.level = level;
+    this.data = data;
+    this.title = title;
+  }
 }
 
 export class NotificationCenter {
   notifications: Notification[];
 
-  constructor() {
+  constructor(public t: ComposerTranslation) {
     this.notifications = [];
+    this.t = t;
   }
 
-  async showModal(
-    title: string,
-    message: string,
-    level: NotificationLevel,
-    addToList: boolean,
+  async showModal({
+    notification,
+    addToList = false,
     trace = false,
-    data = {},
-  ): Promise<void> {
-    const notif: Notification = {
-      id: uuid4(),
-      title: title,
-      message: message,
-      level: level,
-      read: false,
-      time: DateTime.now(),
-      data: data,
-    };
+  }: {
+    notification: Notification,
+    addToList?: boolean,
+    trace?: boolean,
+  }): Promise<void> {
+    if (addToList) {
+      this.addToList(notification);
+    }
     if (trace) {
-      this.trace(notif);
+      this._trace(notification);
     }
-    if (addToList) {
-      this.notifications.push(notif);
+
+    const alertModalConfig: MsAlertModalConfig = {
+      theme: this._getModalTheme(notification.level),
+      subtitle: notification.message,
+      onConfirm: () => {
+        this.markAsRead(notification.id);
```

Comment:
See my remark in MsAlertModal about onConfirm, it would mean that the dismissing is not done here, which is a good thing, it should be left to the modal to decide. It would also mean that the behavior stays consistent between having an `onConfirm` parameter and not having one.